cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -198,6 +199,11 @@ protected long runTask() {
     return connectionInterval;
   }
 
+  private URI appendPath(URI baseUri, String path) {
```

Comment:
is there a benefit to creating the base URI and resolving `newPath` against it, vs having this method take `(String base, String path)` and create the return URI from `newPath`?

Reply:
fixed to return new path from this method. URI resolving is done by the caller